scipy/ndimage/filters.py

```diff
@@ -1186,3 +1186,102 @@ def generic_filter(input, function, size=None, footprint=None,
+        "1. * (np.random.random(shape) > 0.5)")
+    velocity: array_like
+        One velocity vector for each pixel.
+        Must have shape "input.shape + (input.ndim,)"
+        First dimensions are identical to the shape of the random input
+        image, the last dimension defines the coordinate of the velocity.
+    kernel: array_like
+        1-D array with at least on element. Defines the convolution kernel
+        (e.g. a Gaussian kernel constructed with
+        scipy.stats.norm.norm.pdf(np.linspace(-3,3,50)))
+        The flow direction can be visualized by using an asymmetric kernel
+        in combination with an "origin" parameter equal to -len(kernel) // 2
+    origin: int, optional
+        Placement of the filter, by default 0 (which is correct for
+        symmetric kernels). The flow direction can be visualized by using an
+        asymmetric kernel in combination with an "origin" parameter equal to
+        -len(kernel) // 2
+    order: int, optional
+        The order of the spline interpolation used for interpolating the input
+        image and the velocity field, by default 3.
+        See the documentation of
+        "scipy.ndimage.interpolation.map_coordinates" for more details
+    reweight: bool, optional
+        Element-wise reweighting of the filter weights to correct for boundary
+        effects, by default True
+        If set to False, the output image has lower values at the boundary
+    """
+    from scipy.ndimage.interpolation import map_coordinates
+    input = numpy.asarray(input)
+    kernel = numpy.asarray(kernel)
+    velocity = numpy.asarray(velocity)
+    if (kernel.ndim != 1) or (len(kernel) < 1):
+        raise ValueError('Kernel must be 1-D array of length at least 1')
+    if velocity.shape != (input.shape + (input.ndim,)):
+        raise ValueError('Shape of velocity not compatible with shape of '
+                         'input image')
+    if numpy.issubdtype(velocity.dtype, np.floating):
+        float_type = velocity.dtype
+    else:
+        float_type = float
+    center = (len(kernel) // 2) + origin
+    if center < 0:
+        raise ValueError('Origin parameter is too low for the chosen kernel '
+                         'length')
+    if center >= len(kernel):
+        raise ValueError('Origin paremeter is too large for the kernel length')
+    result = kernel[center] * input
+    if reweight:
+        weight = kernel[center] * numpy.ones(input.shape, dtype=float_type)
+    # the kernel is divided into a part left of "center" and a part right of
+    # "center".
+    # The part on the "right" lies in negative flow direction (sign=-1),
+    # the part on the left lies in positive flow direction (sign=1)
+    for sign, ids in [(-1, range(center, len(kernel))),
+                      (+1, reversed(range(0, center+1)))]:
+        # workaround to prevent pyflakes from complaining about undefined
+        # variables
+        v2, m1, v2l22 = None, None, None
```

Comment:
>This will crash for a kernel size of 1 for example.

`i` iterates over `ids`, which in turn is either `range(center, len(kernel))` or `reversed(range(0, center+1))`. The first element of `ids` is always `center`, which means the `else` clause is is not entered in the first iteration.